thorlcr/slave/traceslave.hpp

```diff
@@ -0,0 +1,58 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2015 HPCC Systems.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef TRACESLAVE_HPP
+#define TRACESLAVE_HPP
+
+#include "slave.hpp"
+
+class CTracingThorDataLink : implements CInterface, public IThorDataLink
```

Comment:
We tend to use the CInterfaceOf template in situations like that these days, to avoid double inheritance.
i.e. 
```
class CTracingThorDataLink : public CInterfaceOf<IThorDataLink>
```